dbc2val/Readme.md

```diff
@@ -1,123 +1,366 @@
+      Vehicle.OBD.Speed: 177.12
+
+      -> subscription1:
+      Vehicle.OBD.Speed: 187.44
+
+      -> subscription1:
+      Vehicle.OBD.Speed: 196.64
+
+      -> subscription1:
+      Vehicle.OBD.Speed: 205.28
+      ...
+      Note:
+      To end the subscribiton currently you have to stop the client cli via 'quit' and enter
+   ```
+
+## Steps for a local test with replaying a can dump file
+
+1. Set the a path to a dumpfile e.g. candump.log in the config file `config/dbc_feeder.ini` or use the argument --dumpfile to use a different dumpfile
+
+1. Start the vehicle databroker server
+
+   ```bash
+   cd vehicle_data_broker
+   cargo run --bin vehicle-data-broker
+   Output:
+   Finished dev [unoptimized + debuginfo] target(s) in 0.51s
+   Running `target/debug/vehicle-data-broker`
+   2022-03-04T17:59:01.766338Z  INFO vehicle_data_broker: Init logging from RUST_LOG (environment variable not found)
+   2022-03-04T17:59:01.766403Z  INFO vehicle_data_broker: Starting Vehicle Data Broker
+   2022-03-04T17:59:01.770144Z  INFO vehicle_data_broker: Listening on 127.0.0.1:55555
+   ```
+
+1. Start the vehicle data client cli
+
+   ```bash
+   cd vehicle_data_broker
+   cargo run --bin vehicle-data-cli
+   Output:
+      Finished dev [unoptimized + debuginfo] target(s) in 0.09s
+      Running `target/debug/vehicle-data-cli`
 
-The mapping.yml describes how to map CAN data described in a DBC file to VSS. Let's look at some example entries
+   client> metadata
+      Output:
+      ->
+   ```
 
-```yaml
-PCS_dcdcLvBusVolt:
-  minupdatedelay: 1000
-  targets:
-    Vehicle.OBD.ControlModuleVoltage: {}
+1. Run the dbcfeeder.py
+
+   ```bash
+   cd feeder_can
+   ./dbcfeeder.py
+   ```
+
+1. With the vehicle data client cli
+
+   ```bash
+   client> metadata
+      Output:
+      Vehicle.Powertrain.Transmission.Gear -> id(2)
+      Vehicle.Powertrain.Battery.StateOfCharge.Current -> id(5)
+      Vehicle.Cabin.DogMode -> id(6)
+      Vehicle.OBD.Speed -> id(0)
+      Vehicle.OBD.EngineLoad -> id(1)
+      Vehicle.Cabin.DogModeTemperature -> id(7)
+      Vehicle.Cabin.AmbientAirTemperature -> id(4)
+      Vehicle.Chassis.ParkingBrake.IsEngaged -> id(3)
+
+   client> subscribe SELECT Vehicle.OBD.Speed
+   press 2 times enter
+
+      Example Output:
+      -> subscription1:
+      Vehicle.OBD.Speed: 177.12
+
+      -> subscription1:
+      Vehicle.OBD.Speed: 187.44
+
+      -> subscription1:
+      Vehicle.OBD.Speed: 196.64
+
+      -> subscription1:
+      Vehicle.OBD.Speed: 205.28
+
+      ...
+      Note:
+      To end the subscribiton currently you have to stop the client cli via 'quit' and enter
+   ```
+
+## Provided can-dump files
+
+[candump_Manual_SOC_DogMode_CabinTemp.log](./candump_Manual_SOC_DogMode_CabinTemp.log): Contains state-of-charge, dog mode, and cabin temperature signals. Take this if using the [dog mode example in the vehicle-app-python-sdk repo](https://github.com/SoftwareDefinedVehicle/vehicle-app-python-sdk/tree/main/examples).
```

Comment:
In my opinion, we should not distribute the dog mode exa over too many repos. So, at least later, this would be a candidate to be moved to the dog mode app into Velocitas (Python) SDKs - together with the respective dbc and mapping file. (That would then also help users to understand how to use CAN feeder from another repo.)

This repo should only contain one or two (app unspecific) examples how the feeders work together with Kuksa.VAL server / Kuksa data broker. But this does not necessarily need to happen before going public (but maybe I'm too naive?).

Reply:
I assumed that we would have the dog mode example in, so that the velocitas could use a kuksa.vla.feeeders release. 
But the best way would be extract the dogmode and the velocitas example or who ever should use a different dbc and mapping file. 
And i am also agree the should keep it lean with only a few signals in the mapping file  and they should present in the vss spec.